### PR TITLE
Prepare 2.1 beta release

### LINKED ISSUE #3338: Support for Django 3.0
Hi,

Just a low life dev wondering if you have an expected time frame for Django 3.0 support to be released ? 
I saw #3303 got merged into master a couple month ago. 

Anything I can do to help the delivery process ? 
Should I copy master if I want to use Django 3.0 for the time being ? 

Thanks for all the wonderful work you guys do on this project ! :)

### PR DESCRIPTION
**Pull request by solarissmoke**
Prepare for 2.1 beta release and clean up the release notes and a couple of other errors in the docs.

Closes #3338, closes #3358.

We also need to update the translations, but that is a huge diff so I'll do that after this has been merged.

### CODE REVIEW

File: docs/source/releases/v2.1.rst [hunk 1]
```diff
@@ -15,12 +16,20 @@ Welcome to Oscar 2.1.
 Compatibility
 ~~~~~~~~~~~~~
 
+Oscar 2.1 is compatible with Django versions 2.2 and 3.0, and Python versions 3.5, 3.6 and 3.7.
```
Comment (rik) [inline docs/source/releases/v2.1.rst:19]: Is there any reason 3.8 is not listed here? Seems like the test suites are ok and I couldn't find an issue opened for a compatibility issue.

File: docs/source/releases/v2.1.rst [hunk 2]
```diff
@@ -148,6 +158,11 @@ Removal of deprecated features
   URLs that are not provided by a subclass of ``oscar.core.application.OscarDashboardConfig``
   will result in a ``KeyError``.
 
+- ``customer.forms.PasswordResetForm.get_reset_url`` has been removed.
+
+- The internal and undocumented class ``oscar.core.compat.UnicodeSCVReader``
```
Comment (rik) [inline docs/source/releases/v2.1.rst:163]:
```suggestion
- The internal and undocumented class ``oscar.core.compat.UnicodeCSVReader``
```

File: docs/source/releases/v2.1.rst (hunk 1 above)
Comment (solarissmoke) [reply docs/source/releases/v2.1.rst:19]: No, an omission on my part. I'll add it.

**Comment by fceruti** [comment]
Hey guys, thanks for the great work. I don't know where to put this, or where to interact with you, but I wanted to know if #2730 is going to make it for this release. If you don't have the manpower, I can help with that. I'm just not sure where to start.

**Comment by solarissmoke** [comment]
@fceruti no, there will be a separate major release in a few weeks with that. See #3304

**Comment by solarissmoke** [comment]
@django-oscar/core would be grateful if any of you has a few minutes to review this - it's just a change to release notes so that I can cut a beta release.

**Pull request merged**